designer/server/src/main/scala/pl/touk/nussknacker/ui/process/repository/ScenarioActionRepositoryAuditLogDecorator.scala

```diff
@@ -0,0 +1,177 @@
+package pl.touk.nussknacker.ui.process.repository
+
+import db.util.DBIOActionInstances._
+import pl.touk.nussknacker.engine.api.Comment
+import pl.touk.nussknacker.engine.api.deployment.ProcessActionState.ProcessActionState
+import pl.touk.nussknacker.engine.api.deployment._
+import pl.touk.nussknacker.engine.api.process.{ProcessId, ProcessName, ProcessingType, VersionId}
+import pl.touk.nussknacker.ui.process.ScenarioActivityAuditLog
+import pl.touk.nussknacker.ui.security.api.LoggedUser
+
+import java.time.Instant
+import scala.concurrent.ExecutionContext
+
+class ScenarioActionRepositoryAuditLogDecorator(underlying: ScenarioActionRepository)(
+    implicit executionContext: ExecutionContext
+) extends ScenarioActionRepository {
+
+  override def addInstantAction(
+      processId: ProcessId,
+      processVersion: VersionId,
+      actionName: ScenarioActionName,
+      comment: Option[Comment],
+      buildInfoProcessingType: Option[ProcessingType]
+  )(implicit user: LoggedUser): DB[ProcessAction] =
+    underlying
+      .addInstantAction(processId, processVersion, actionName, comment, buildInfoProcessingType)
+      .map { processAction =>
+        ScenarioActivityAuditLog.onScenarioImmediateAction(
+          processAction.id,
+          processId,
+          actionName,
+          Some(processVersion),
+          user
+        )
+        processAction
+      }
+
+  override def addInProgressAction(
+      processId: ProcessId,
+      actionName: ScenarioActionName,
+      processVersion: Option[VersionId],
+      buildInfoProcessingType: Option[ProcessingType]
+  )(implicit user: LoggedUser): DB[ProcessActionId] =
+    underlying
+      .addInProgressAction(processId, actionName, processVersion, buildInfoProcessingType)
+      .map { processActionId =>
+        ScenarioActivityAuditLog.onScenarioActionStarted(processActionId, processId, actionName, processVersion, user)
+        processActionId
+      }
+
+  override def markActionAsFinished(
+      actionId: ProcessActionId,
+      processId: ProcessId,
+      actionName: ScenarioActionName,
+      processVersion: VersionId,
+      performedAt: Instant,
+      comment: Option[Comment],
+      buildInfoProcessingType: Option[ProcessingType]
+  )(implicit user: LoggedUser): DB[Unit] =
+    underlying
+      .markActionAsFinished(
+        actionId,
+        processId,
+        actionName,
+        processVersion,
+        performedAt,
+        comment,
+        buildInfoProcessingType
+      )
+      .map { _ =>
```

Comment:
maybe we could add some `onSuccess` (or `andThen`) util to avoid mapping, side-effecting, and returning the result all over again?